plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -46,22 +46,70 @@ export const Reader = ({ entityId, onReady }: Props) => {
   const theme = useTheme<BackstageTheme>();
 
   const techdocsStorageApi = useApi(techdocsStorageApiRef);
-  const [shadowDomRef, shadowRoot] = useShadowDom();
   const navigate = useNavigate();
+  const shadowDomRef = useRef(null);
+  const [loadedPath, setLoadedPath] = useState('');
+  const [atInitialLoad, setAtInitialLoad] = useState(true);
+  const [newerDocsExist, setNewerDocsExist] = useState(false);
 
-  const { value, loading, error } = useAsync(async () => {
+  const {
+    value: isSynced,
+    loading: syncInProgress,
+    error: syncError,
+  } = useAsync(async () => {
+    if ((await techdocsStorageApi.getBuilder()) !== 'local') {
```

Comment:
Just adding a comment here to explain why

```suggestion
  } = useAsync(async () => {
    // Attempt to sync only if `techdocs.builder` in app config is set to 'local'
    if ((await techdocsStorageApi.getBuilder()) !== 'local') {
```